Starting multilingual post (#440)
ropensci/roweb3: content/blog/2022-11-29-multilingual-publishing-en/index.md

---
title: Expanding our Community through Multilingual Publishing 
author:
  - Yanina Bellini Saibene
  - Paola Corrales
  - Elio Campitelli
  - Maëlle Salmon
date: '2022-11-29'
slug: multilingual-publishing-en
tags:
  - community
  - multilingual
description: "We are translating rOpenSci's materials on best practice for software development, code review, and contribution to open source projects into Spanish! We are also developing guidelines and tools for translating open source resources to reach a wider audience. Learn about the project in this blog post."

---

We are excited to announce that, with the support of the Chan Zuckerberg Initiative, NumFOCUS, and the R Consortium, we have begun the process of translating rOpenSci's materials on best practice for software development, code review, and contribution to open source projects into Spanish. 
As part of this effort, we are also developing guidelines and tools for translating open source resources to reach a wider audience.

> According to a recent report from The Linux Foundation, _language access_ is one of the _environmental barriers to equity in open source_. The report states that 

Reviewer: ```suggestion
According to a [recent report](https://www.linuxfoundation.org/research/the-2021-linux-foundation-report-on-diversity-equity-and-inclusion-in-open-source) from The Linux Foundation, _language access_ is one of the _environmental barriers to equity in open source_. The report states that 
```

> Do you even know how smart I am in Spanish?

By publishing multilingual resources, we can lower these barriers by increasing access to knowledge and democratizing access to quality resources, thereby increasing the potential for individuals to contribute to software and open science projects.

## Language representation

The Spanish-speaking R community has been very active and growing in recent years, and has undertaken various translation activities for technical materials such as books, cheatsheets, guides, and datasets.

In 2017, several Latin American R-Ladies began translating the R-Ladies' [Code of Conduct](https://github.com/rladies/starter-kit/wiki/Code-of-Conduct#spanish) and their Rules and Guidelines into Spanish.

In 2018, the [R community in Latin America collectively translated](https://youtu.be/kuHjpdgkSxg) the [R for Data Science](https://es.r4ds.hadley.nz/) book into Spanish. This included the translation of all the data sets used in the book, which were compiled in the [datos](https://github.com/cienciadedatos/datos) package, making it an excellent tool for teaching.

> The community continued with the translation of [Teaching Tech Together](http://teachtogether.tech/es/index.html) and contributed to the translations into Spanish of the [Posit Cheatsheets](https://posit.co/resources/cheatsheets/?type=translations/#translation-12),

Reviewer: ```suggestion
The community continued with the translation of [Teaching Tech Together](https://teachtogether.tech/es/index.html) and contributed to the translations into Spanish of the [Posit Cheatsheets](https://posit.co/resources/cheatsheets/?type=translations/#translation-12),
```

[The Carpentries](https://software-carpentry.org/lessons/)’ 
and [The Programming Historian](https://programminghistorian.org/es/) lessons.

> Driven by this active and growing Spanish-speaking community, rOpenSci [successfully piloted](https://ropensci.org/blog/2021/07/27/censo2017/) our first [Spanish-language peer review](https://ropensci.org/commcalls/2019-06-28/), where the submission itself, as well as all reviews and editorial responses, were in Spanish.

Reviewer: ```suggestion
Driven by this active and growing Spanish-speaking community, rOpenSci [successfully piloted](/blog/2021/07/27/censo2017/) our first [Spanish-language peer review](/commcalls/2019-06-28/), where the submission itself, as well as all reviews and editorial responses, were in Spanish.
```

rOpenSci provides community support, standards, and infrastructure for scientists 
and research software engineers working in R to develop, maintain, and publish high-quality open-source scientific software.
In addition, we develop and maintain high-quality documentation and [resources](/resources/) to support these activities.  

Miltilingual Publishing of documentation and resources involve two aspects:  

* The _internationalization_ refers to the technological solution allowing the software to adapt to several regions without engineering changes to the source code. It is what allow us to localize our content[^3].

* The _localization_ is when you take a piece of content in its original form and convert it into something more accessible and suitable for another region, country, or audience [^3]. Includes language, date formats, currency, measure units, and support for different characters.

Besides, technical translations of living documents have two well-defined stages involving different resources: 

  1. achieving a first version of the translated material and

  2. keeping the material updated and synchronized between the different languages.

> There are many solutions for internationalizing and localizing content and software. _Translation_ is typically the most time-consuming component of these efforts. [^3]

Reviewer 2: ```suggestion
There are many solutions for internationalizing and localizing content and software. _Translation_ is typically the most time-consuming component of these efforts[^3]. 
```

Based on the experience gain on the previews projects and the characteristics of our community, this project aims to advance in stage 1 achieving a first revised translation in Spanish of the rOpenSci materials, and testing a technical workflow to perform the translations.

> Our approach will test to generate a workflow using an API of an automatic translator to generate a first translated version of the documents that will be later reviewed by professional bilingual developers using GitHUb as platform.

Author: Yes.

Reviewers will focus on linguistic tasks such as the use of inclusive language, the localization of examples, and the correct translation of phrases, metaphors, or analogies. 
They will also look for reference material in the language of the translation.
In addition, if the workflow proves feasible and useful, it could be incorporated into the pipeline to keep the texts in different languages synchronized, contributing to solve stage 2.

Besides systematizing the experiences, other products of the project will be:

* a general guide of rules to follow when translating technical documents,
* a particular guide for the Spanish language,

> * a list of technical terms that are translated into Spanish and those that are not,

Reviewer: ```suggestion
* a list of technical terms which will be translated into Spanish and a list of those that will not be,
```

* an R package with our workflow for translating technical documents.

> ## Building Community

Reviewer: ```suggestion
## Building community
```

Multilingual publishing will help democratize access to quality resources for learning and teaching the necessary digital skills to become contributors to open source projects and to have a workflow for performing computationally reproducible open science.

At rOpenSci, we understand review as a way of building community.

> The material for our review process in Spanish will allow rOpenSci to continue building the community in Spanish-speaking regions, such as Latin America, to increase the number of contributors and get feedback on how our tools and processes can be improved to better serve these community members.

Reviewer: ```suggestion
Developing the material for our review process in Spanish will allow rOpenSci to continue building the community in Spanish-speaking regions (such as Latin America), to increase the number of contributors, and to get feedback on how our tools and processes can be improved to better serve these community members.
```

---
title: Expanding our Community through Multilingual Publishing 
author:
  - Yanina Bellini Saibene
  - Paola Corrales
  - Elio Campitelli
  - Maëlle Salmon
date: '2022-11-29'
slug: multilingual-publishing-en
tags:

>   - community

Author: mmm...yes!

As part of this effort we are also developing guidelines and tools for translating open source resources to a wider audience.

## Language representation

Approximately 64% of Internet content is in English, although only 4.8% of the world's population speaks English as a first language. For this reason, we are embarking on an effort to translate rOpenSci’s material on best practices for software development, code review, and contribution to open source projects beyond the English language starting with Spanish.

Spanish is the second most-spoken native language in the world. It is also one of the most geographically widespread languages, reaching many countries with Spanish as an official language. 

The Spanish-speaking R Community has been growing and very active in the last yearsand have carried out various activities related to translations of materials such as books, cheatshets, guides and datasets. 

In 2017 several Latin American R-Ladies start with the translations to Spanish of the R-Ladies' [Code of Conduct](https://github.com/rladies/starter-kit/wiki/Code-of-Conduct#spanish) and their Rules and Guidelines. 

> In 2018, the R Community in Latin America collectively translated the [R for Data Science (R4DS) book](https://es.r4ds.hadley.nz/) to Spanish.  This includes the translation of the package [datos](https://github.com/cienciadedatos/datos) to Spanish and dados to [Portuguese](https://cran.r-project.org/web/packages/dados/index.html), becoming an excellent tool for teaching. From the success of the collective and volunteer translation of R4DS, the community also translated [Teaching Tech Together](http://teachtogether.tech/es/index.html) and contributed to several translations to Spanish of [RStudio CheatSheets](https://posit.co/resources/cheatsheets/?type=translations/#translation-12), [The Carpentries](https://software-carpentry.org/lessons/)’ and [The Programming Historian](https://programminghistorian.org/es/) lessons. 

Reviewer 2: There isn 't a package in English, the package datos is a collection of data sets spread through different R packages like ggplot2. Maybe we can refrase it as:

```suggestion
In 2018, the R Community in Latin America collectively translated the [R for Data Science (R4DS) book](https://es.r4ds.hadley.nz/) to Spanish.  This includes the translation of all the data sets included in the book compiled in the package [datos](https://github.com/cienciadedatos/datos) in Spanish and dados in [Portuguese](https://cran.r-project.org/web/packages/dados/index.html), becoming an excellent tool for teaching. From the success of the collective and volunteer translation of R4DS, the community also translated [Teaching Tech Together](http://teachtogether.tech/es/index.html) and contributed to several translations to Spanish of [RStudio CheatSheets](https://posit.co/resources/cheatsheets/?type=translations/#translation-12), [The Carpentries](https://software-carpentry.org/lessons/)’ and [The Programming Historian](https://programminghistorian.org/es/) lessons. 
```

> Driven by this active and growing spanish-speaking community, rOpenSci [successfully piloted](https://ropensci.org/blog/2021/07/27/censo2017/) our first [Spanish-language peer review](https://ropensci.org/commcalls/2019-06-28/), where the submission itself, and all reviews, and editorial responses were in Spanish.

Reviewer 3: ```suggestion
Driven by this active and growing Spanish-speaking community, rOpenSci [successfully piloted](https://ropensci.org/blog/2021/07/27/censo2017/) our first [Spanish-language peer review](https://ropensci.org/commcalls/2019-06-28/), where the submission itself, and all reviews, and editorial responses were in Spanish.
```

All this work and previous experience generated the right conditions to start our multilingual publishing with Spanish.

## Technical materials, technical solutions

> rOpenSci provides community support, standards, and infrastructure for scientists and research software engineers working in R to develop, maintain, and publish high-quality open-source scientific software. In addition, we develop and maintain high-quality documentation and resources to support these activities.  

Reviewer 3: ```suggestion
rOpenSci provides community support, standards, and infrastructure for scientists and research software engineers working in R to develop, maintain, and publish high-quality open-source scientific software. In addition, we develop and maintain high-quality documentation and [resources](/resources/) to support these activities.  
```

> We publish our material using technical documents which are rendered using software. The translations of technical documents have two aspects (a) the internationalization (i18n) that provides the framework to support translation and requires technical knowledge (need to deal with source code vs. rendered version of the material), and (b) the localization (l10n), which is the task of translating the text and requires linguistic knowledge. 

Reviewer 3: could the part in parenthesis be removed?

Also, in our case, translating the text _also_ require technical knowledge, but on the topic of the content, not on the topic of R Markdown / Quarto /etc.

> At the same time, technical translations of living documents have two well-defined stages involving different resources:  (a) achieving a first version of the translated material and (b) keeping the material updated and synchronized between the different languages.

Reviewer 3: ```suggestion
Besides, technical translations of living documents have two well-defined stages involving different resources:  (a) achieving a first version of the translated material and (b) keeping the material updated and synchronized between the different languages.
```

> There are several approaches and technologies to work on this task. Based on the experience gain on the mentions previews projects this project aims to advance in stage (a) achieving a first revised translation in Spanish of the rOpenSci materials and testing two different technical workflows to perform the translations.

Reviewer 3: ```suggestion
There are several approaches and technologies to work on translations. Based on the experience gain on the mentions previews projects this project aims to advance in stage (a) achieving a first revised translation in Spanish of the rOpenSci materials and testing two different technical workflows to perform the translations.
```

> The first workflow will use GitHub and GitHub review as a platform for translations and revisions since our materials are on this platform, and it has already been used in other translation projects.  

Reviewer 3: but in the end we are testing one workflow only? Which is a combination of both things described here?

The second approach will test to generate a workflow using an API of an automatic translator to generate a first translated version of the documents that will be later reviewed by professional bilingual developers also using GitHUb as platform.

> The advantage of the second workflow is to reduce the time it takes to produce a quality translation.  Reviewers will focus on linguistic tasks such as the use of inclusive language, the localization of examples, and the correct translation of phrases, metaphors, or analogies. They will also look for references in the language of the translation. In addition, if the workflow proves feasible and useful, it could be incorporated into the pipeline to keep the texts in different languages synchronized, contributing to solve stage (b).

Reviewer 3: ```suggestion
The advantage of the second workflow is to reduce the time it takes to produce a quality translation.  Reviewers will focus on linguistic tasks such as the use of inclusive language, the localization of examples, and the correct translation of phrases, metaphors, or analogies. They will also look for references in the language of the translation. In addition, if the workflow proves feasible and useful, it could be incorporated into the pipeline to keep the texts in different languages synchronized, contributing to solve stage (b).
```

Maybe clarify what "references" mean here?

> Besides systematizing both experiences, other products of the project will be: (a) a general guide of rules to follow when translating technical documents, (b) a particular guide for the Spanish language, (c) a list of technical terms that are translated and those that are not, (d) an R package with our workflow to translate technical documents. 

Reviewer 3: ```suggestion
Besides systematizing both experiences, other products of the project will be: (a) a general guide of rules to follow when translating technical documents, (b) a particular guide for the Spanish language, (c) a list of technical terms that are translated to Spanish and those that are not, (d) an R package with our workflow to translate technical documents. 
```

## Building Community

> Translating rOpenSci material to Spanish will help democratize access to these quality resources to learn and teach the necessary digital skills to become contributors to Open Source projects not only because you can now learn how to be a developer, a maintainer, a reviewer or an editor, but also because translation is often a good first way to contribute and is recognized as valuable by the community. This material is also useful for learning and teaching how to have a workflow to perform computationally reproducible Open Science.

Reviewer 3: ```suggestion
Translating rOpenSci material to Spanish will help democratize access to these quality resources to learn and teach the necessary digital skills to become contributors to Open Source projects not only because you can now learn how to be a developer, a maintainer, a reviewer or an editor; but also because translation is often a good first way to contribute, and is recognized as valuable by the community. This material is also useful for learning and teaching how to have a workflow to perform computationally reproducible Open Science.
```

---
title: Expanding our Community through Multilingual Publishing 
author:
  - Yanina Bellini Saibene
  - Paola Corrales
  - Elio Campitelli
  - Maëlle Salmon
date: '2022-11-29'
slug: multilingual-publishing-en
tags:
  - community
  - multilingual

> description: ""

Reviewer 3: don't forget to fill this :-)

---

> We are excited to announce that, with the support of the Chan-Zuckerberg Initiative, NumFOCUS, and the R Consortium, we have begun the process of translating rOpenSci's materials on best practices for software development, code review, and contribution to open source projects into Spanish. As part of this effort, we are also developing guidelines and tools for translating open source resources to reach a wider audience.

Reviewer 3: ```suggestion
We are excited to announce that, with the support of the Chan Zuckerberg Initiative, NumFOCUS, and the R Consortium, we have begun the process of translating rOpenSci's materials on best practice for software development, code review, and contribution to open source projects into Spanish. 
As part of this effort, we are also developing guidelines and tools for translating open source resources to reach a wider audience.
```

According to a recent report from The Linux Foundation, _language access_ is one of the _environmental barriers to equity in open source_. The report states that 

> The predominant language of open source is English—in code, content, and community interactions—and English proficiency is a metric by which performance and personality can be judged[^1]. 

By publishing multilingual resources, we can lower these barriers by increasing access to knowledge and democratizing access to quality resources, thereby increasing the potential for individuals to contribute to software and open science projects.

> ## Language representation

Reviewer 3: could some of the paragraphs in this section be merged?

> The Spanish-speaking R community has been very active and growing in recent years, and has undertaken various translation activities for technical materials such as books, cheat sheets, guides, and datasets.

Reviewer 3: ```suggestion
The Spanish-speaking R community has been very active and growing in recent years, and has undertaken various translation activities for technical materials such as books, cheatsheets, guides, and datasets.
```

since that's how Posit writes it

In 2017, several Latin American R-Ladies began translating the R-Ladies' [Code of Conduct](https://github.com/rladies/starter-kit/wiki/Code-of-Conduct#spanish) and their Rules and Guidelines into Spanish.

> In 2018, the R community in Latin America collectively translated the [R for Data Science](https://es.r4ds.hadley.nz/) book into Spanish. This included the translation of all the data sets used in the book, which were compiled in the [datos](https://github.com/cienciadedatos/datos) package, making it an excellent tool for teaching.

Reviewer 3: is there a recording of a talk about this project? I vaguely remember there was one at a LatinR conference but maybe I'm wrong?

Author: Yes! It is a talk by Riva on RStudioConf: https://youtu.be/kuHjpdgkSxg
She also talks at useR! 2019 but no recording of that one.

> In 2018, the R community in Latin America collectively translated the [R for Data Science](https://es.r4ds.hadley.nz/) book into Spanish. This included the translation of all the data sets used in the book, which were compiled in the [datos](https://github.com/cienciadedatos/datos) package, making it an excellent tool for teaching.

Author: ```suggestion
In 2018, the [R community in Latin America collectively translated](https://youtu.be/kuHjpdgkSxg) the [R for Data Science](https://es.r4ds.hadley.nz/) book into Spanish. This included the translation of all the data sets used in the book, which were compiled in the [datos](https://github.com/cienciadedatos/datos) package, making it an excellent tool for teaching.
```

> The community continued with the translation of [Teaching Tech Together](http://teachtogether.tech/es/index.html) and contributed to the translations into Spanish of the [Posit CheatSheets](https://posit.co/resources/cheatsheets/?type=translations/#translation-12),

Reviewer 3: ```suggestion
The community continued with the translation of [Teaching Tech Together](http://teachtogether.tech/es/index.html) and contributed to the translations into Spanish of the [Posit Cheatsheets](https://posit.co/resources/cheatsheets/?type=translations/#translation-12),
```

All of this work and previous experience created the right conditions for us to start our multilingual publishing with _Spanish_. 
Spanish is the second most-spoken native language in the world and is one of the most geographically widespread languages, being an official language in many countries. [^2]

## Technical materials, technical solutions

rOpenSci provides community support, standards, and infrastructure for scientists and research software engineers working in R to develop, maintain, and publish high-quality open-source scientific software. 
In addition, we develop and maintain high-quality documentation and [resources](/resources/) to support these activities.

Multilingual publishing of documentation and resources involves two aspects:

* The _internationalization_ refers to the technological solution allowing the software to adapt to different regions without requiring engineering changes to the source code. It is what enables us to localize our content[^3].

> * The _localization_ is the process of taking a piece of content in its original form and converting it into something more accessible and suitable for another region, country, or audience [^3]. This includes language, date formats, currency, measurement units, and support for different character sets. 

Reviewer 3: ```suggestion
* The _localization_ is the process of taking a piece of content in its original form and converting it into something more accessible and suitable for another region, country, or audience[^3]. This includes language, date formats, currency, measurement units, and support for different character sets. 
```

In addition, technical translations of living documents have two well-defined stages that involve different resources:

  1. achieving a first version of the translated material, and
  
  2. keeping the material updated and synchronized between the supported languages.

There are many solutions for internationalizing and localizing content and software. _Translation_ is typically the most time-consuming component of these efforts[^3].

> Based on the experience gained from out pilot projects and the characteristics of our community, this project aims to progress to stage 1 by achieving a first revised translation in Spanish of the rOpenSci materials, and testing a technical workflow for performing the translations.

Reviewer 3: ```suggestion
Based on the experience gained from our pilot projects and the characteristics of our community, this project aims to progress to stage 1 by achieving a first revised translation in Spanish of the rOpenSci materials, and testing a technical workflow for performing the translations.
```

are these pilot projects or "first experiences"?

> Our approach will test and develop a workflow using an API of an automatic translator to generate a first translated version of the documents, which will be later reviewed by professional bilingual developers using GitHub as a platform.

Reviewer 3: why not mention DeepL?

---
title: Expanding our Community through Multilingual Publishing 
author:
  - Yanina Bellini Saibene
  - Paola Corrales
  - Elio Campitelli
  - Maëlle Salmon
date: '2022-11-29'
slug: multilingual-publishing-en
tags:
  - community
  - multilingual
description: ""

---

We are excited to announce that, with the support of the Chan Zuckerberg Initiative, NumFOCUS, and the R Consortium, we have begun the process of translating rOpenSci's materials on best practice for software development, code review, and contribution to open source projects into Spanish. 
As part of this effort, we are also developing guidelines and tools for translating open source resources to reach a wider audience.

According to a recent report from The Linux Foundation, _language access_ is one of the _environmental barriers to equity in open source_. The report states that 

> The predominant language of open source is English—in code, content, and community interactions—and English proficiency is a metric by which performance and personality can be judged[^1]. 

> 

Reviewer 3: ```suggestion

This might remind some readers of the famous [quote](https://www.imdb.com/title/tt4166966/characters/nm0005527) from [Modern Family](https://www.tiktok.com/@primevideouk/video/6899066588121713921), where a character called Gloria exclaims:

> Do you even know how smart I am in Spanish?

```

Reviewer 3: @yabellini just to be sure we're not infringing copyrights, I am linking out to it but maybe I'm being overly cautious.

In addition, we develop and maintain high-quality documentation and [resources](/resources/) to support these activities.

Multilingual publishing of documentation and resources involves two aspects:

* The _internationalization_ refers to the technological solution allowing the software to adapt to different regions without requiring engineering changes to the source code. It is what enables us to localize our content[^3].

* The _localization_ is the process of taking a piece of content in its original form and converting it into something more accessible and suitable for another region, country, or audience[^3]. This includes language, date formats, currency, measurement units, and support for different character sets. 

In addition, technical translations of living documents have two well-defined stages that involve different resources:

  1. achieving a first version of the translated material, and
  
  2. keeping the material updated and synchronized between the supported languages.

There are many solutions for internationalizing and localizing content and software. _Translation_ is typically the most time-consuming component of these efforts[^3].

> Based on the experience gained from our pilot projects and the characteristics of our community, this project aims to progress to stage 1 by achieving a first revised translation in Spanish of the rOpenSci materials, and testing a technical workflow for performing the translations.

Author: ```suggestion
Based on the experience gained from our first experiences and the characteristics of our community, this project aims to progress to stage 1 by achieving a first revised translation in Spanish of the rOpenSci materials, and testing a technical workflow for performing the translations.
```

> Our approach will test and develop a workflow using an API of an automatic translator to generate a first translated version of the documents, which will be later reviewed by professional bilingual developers using GitHub as a platform.

Author: ```suggestion
Our approach will test and develop a workflow using the automatic translator DeepL API to generate a first translated version of the documents, which will be later reviewed by professional bilingual developers using GitHub as a platform.
```
